infrastructure/controller/pkg/endpoints/components_update.go

```diff
@@ -71,7 +69,10 @@ func (s *ComponentsUpdate) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 //NOTE: Prevent the upload of a configmap if the component is not present
 func (s *ComponentsUpdate) isComponentInstalled(configName string) bool {
-	deployedCharts := s.DeployedCharts.GetDeployedCharts()
-
+	deployedCharts, err := k8s.GetInstalledComponents(context.TODO(), s.namespace, s.clientSet)
+	if err != nil {
+		klog.Error("Unable to get installed components:\n" + err.Error())
```

Comment:
In the case of an error here the `/components.update` call should fail though, right? It means that there is something not right with the cluster. 

In any case this function should return the error so that we can differentiate between both cases 🤔

Reply:
Ok, in that case I will make the function return `bool,err` :+1: